### PR TITLE
Allow deleting storage pools that only contain image volumes

### LINKED ISSUE #4792: Allow deleting storage pools that only contain image volumes
It's a bit annoying having to go and manually delete every optimized image from a storage pool when the user didn't have to do anything to put them there.

So instead, we should just have the storage pool deletion code wipe those for the user as needed, effectively only refusing deletion of the pool if anything else is using it.

### PR DESCRIPTION
**Pull request by monstermunchkin**
Closes #4792

### CODE REVIEW

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by brauner** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
I've reset CEPH, so lets see if this works now.

**Comment by stgraber** [comment]
jenkins: test this please

**Comment by stgraber** [comment]
jenkins: test this please

File: lxd/storage_pools.go [hunk 1]
```diff
@@ -557,6 +558,57 @@ func storagePoolDelete(d *Daemon, r *http.Request) Response {
...
+			fname = shared.VarPath("images", imgInfo.Fingerprint) + ".rootfs"
+			if shared.PathExists(fname) {
+				err = os.Remove(fname)
+				if err != nil {
+					logger.Debugf("Error deleting image file %s: %s", fname, err)
+				}
+			}
+
+			// Remove the database entry for the image.
+			d.cluster.ImageDelete(imgID)
```
Comment (stgraber) [inline lxd/storage_pools.go]: This is pretty wrong here and the reason why the CEPH test fails.

We need to only remove the image from the pool we're deleting, we very much MUST keep the image in the database and in the "images" directory.

**Comment by monstermunchkin** [comment]
The images aren't deleted anymore. This should fix the Ceph issue.

File: lxd/storage_pools.go [hunk 2]
```diff
@@ -557,6 +557,23 @@ func storagePoolDelete(d *Daemon, r *http.Request) Response {
 		return EmptySyncResponse
 	}
 
+	volumeNames, err := d.cluster.StoragePoolVolumesGetNames(poolID)
```
Comment (brauner) [inline lxd/storage_pools.go]: Hm, what happens if there are non image volumes still existent? Would this still delete all images and then fail?

Comment (monstermunchkin) [reply lxd/storage_pools.go]: If there are non-image-volumes existent, this code won't be reached but fail when calling `storagePoolDeleteCheckPreconditions`.

Comment (brauner) [reply lxd/storage_pools.go]: Ok

**Pull request merged**